### PR TITLE
e2e test for dbos-demo-apps

### PR DESCRIPTION
**Pull request by CaspianA1**

### CODE REVIEW

File: e2e_test.bash
Comment (qianl15) [inline e2e_test.bash:1]: I'll rename this file to `e2e_test.sh` -- `.sh` is a more commonly used file extension.

**Review by qianl15 — APPROVED** [review]
One minor note on the file name, otherwise looks good!

File: e2e_test.bash [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+log() {
+  echo -e "\n>>>>> $1\n"
```
Comment (kraftp) [inline e2e_test.bash]: To make this easier to debug, use `set -ex` instead of `try_command` so it prints each of its commands and fails entirely on the first command failure

Comment (kraftp) [inline e2e_test.bash]: Also why is this file here twice?

File: e2e_test.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+####################################################################################################
+
+# This list should be extended as more demos are written!
+directories=(
+  bank/bank-backend
+  # bank/bank-frontend
+
+  e-commerce/payment-backend
+  e-commerce/shop-backend
+  e-commerce/shop-frontend
```
Comment (kraftp) [inline e2e_test.sh:17]: Not a DBOS app

File: e2e_test.bash (hunk 1 above)
Comment (CaspianA1) [reply e2e_test.bash]: Just fixed this

**Review by kraftp — COMMENTED** [review]
Looks good, but you can simplify the bash script significantly

File: e2e_test.sh (hunk 2 above)
Comment (CaspianA1) [reply e2e_test.sh:17]: Would still be good to test the linter here, regardless, I think

**Review by chuck-dbos — APPROVED** [review]

File: e2e_test.bash
Comment (CaspianA1) [reply e2e_test.bash:1]: Done

File: e2e_test.bash (hunk 1 above)
Comment (CaspianA1) [reply e2e_test.bash]: Done

**Review by kraftp — APPROVED** [review]
